elephant/spike_train_synchrony.py

```diff
@@ -341,33 +342,37 @@ def delete_synchrofacts(self, threshold, in_place=False, mode='delete'):
             if mode == 'extract':
                 mask = np.invert(mask)
             new_st = st[mask]
-            spiketrain_list[idx] = new_st
-            if in_place:
+            if in_place and st.segment is not None:
                 segment = st.segment
-                if segment is None:
-                    continue
 
-                # replace link to spiketrain in segment
-                new_index = self._get_spiketrain_index(
-                    segment.spiketrains, st)
-                segment.spiketrains[new_index] = new_st
+                try:
+                    # replace link to spiketrain in segment
+                    new_index = self._get_spiketrain_index(
+                        segment.spiketrains, st)
+                    segment.spiketrains[new_index] = new_st
+                except ValueError:
+                    # st is not in this segment even though it points to it
+                    warnings.warn(f"The SpikeTrain at index {idx} of the "
+                                  "input list spiketrains has a "
+                                  "unidirectional uplink to a segment in "
+                                  "whose segment.spiketrains list it does not "
+                                  "appear.")
 
                 block = segment.block
-                if block is None:
-                    continue
-
-                # replace link to spiketrain in groups
-                for group in block.groups:
-                    try:
-                        idx = self._get_spiketrain_index(
-                            group.spiketrains,
-                            st)
-                    except ValueError:
-                        # st is not in this group, move to next group
-                        continue
-
-                    # st found in group, replace with new_st
-                    group.spiketrains[idx] = new_st
+                if block is not None:
+                    # replace link to spiketrain in groups
+                    for group in block.groups:
+                        try:
+                            idx = self._get_spiketrain_index(
+                                group.spiketrains,
+                                st)
+                        except ValueError:
+                            # st is not in this group, move to next group
+                            continue
+
+                        # st found in group, replace with new_st
+                        group.spiketrains[idx] = new_st
```

Comment:
this change is not necessary but OK
you've just increased the number of tabs --> increased (insignificantly) the code complexity

Reply:
It is necessary because the `continue` was skipping the line below the block you commented on here which replaces the spiketrain in the input list.